ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,858 @@
+
+Alternatively, `IntermediateCatchEvent:Event_Occurred` could be added to the processing result of `WorkflowInstanceSubscription:Correlate`.
+
+### Interrupting Timer Boundary Event
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+ServiceTask:Element_Activating | ServiceTask:Element_Activated, Timer:Created, Job:Created
+Timer:Trigger | Timer:Triggered, ServiceTask:Element_Terminating
+ServiceTask:Element_Terminating | ServiceTask:Element_Terminated, BoundaryEvent:Element_Activating 
+BoundaryEvent:Element_Activating | BoundaryEvent:Element_Activated, BoundaryEvent:Element_Completing
+
+* on processing the command `Timer:Trigger`, we store the element which was triggered in the state 
+* on processing the command `ServiceTask:Element_Terminating`, we check the state for triggered events and activate the triggered event
+
+Alternatively, `BoundaryEvent:Event_Occurred` could be added to the processing result of `Timer:Trigger`.
+
+### Boundary Event
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+... | ...
+
+Concurrent commands for the service task and the boundary event:
+
+![boundary-event](ZEP-wf-stream-processing/event-sourcing-boundary-event-1.png)
+
+* the timer is triggered before the job is completed
+
+![boundary-event](ZEP-wf-stream-processing/event-sourcing-boundary-event-2.png)
+
+* the job is completed before the timer is triggered
+* TODO: reject message correlate command (rejection is relevant for the correct correlation) 
+
+
+### Event-Sub-Process
+
+...
+
+### Message Start Event
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+MessageStartEventSubscription:Open | MessageStartEventSubscription:Opened
+Message:Publish | Message:Published, StartEvent:Event_Occurred, Process:Element_Activating
+
+* WorkflowInstanceSubscription is not necessary like in message intermediate catch event, because a new workflow instance can be created on any partition
+* `MessageStartEventSubscription` is currently not correlated, but this could be considered a business logic change for after the implementation of this proposal
+* instead of `Process:Element_Activating` in response to `Message:Publish`, we could also consider writing `WorkflowInstanceCreationIntent:Create` to create a workflow instance. This can also be considered a business logic change for after the implementation of this proposal
+
+### Multi-Instance Activity
+ 
+For an input collection of 2 elements:
+
+#### Sequential Multi-Instance
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+MultiInstanceBody:Element_Activating | MultiInstanceBody:Element_Activated, Variable:Created (output collection), Variable:Created (input element), Variable:Created (output element), Variable:Created (loop counter), ServiceTask:Element_Activating
```

Comment:
Could be interesting in the future to have a way to set multiple variables in a single record, e.g. `VariableBatch` or something (not a good name, but taking inspiration from `JobBatch` :sweat_smile:), to reduce overhead of writing multiple records atomically.